### DISCUSSION TITLE
Serialization error in entity with UserInterface

### DISCUSSION THREAD

**Discussion by szal1k** [rekalogika/file-src#53] [Q&A] [answered]
Hi,
I was added "avatar" field to my sample "User" entity with UserInterface and standard PasswordAuthenticatedUserInterface. After upload file i get an error from "Rekalogika\File\File" function "__serialize": 

"Serialization is not supported. Use getPointer() to get a pointer to this file and serialize it instead."

The solution is add custom serialization to User entity (without "avatar" FileInterface field) as:
`
class User implements UserInterface, PasswordAuthenticatedUserInterface, PersistenceSubjectInterface, \Serializable
{
...
    public function serialize()
    {
        return serialize(array(
            'id' => $this->id,
            'email' => $this->email,
            'password' => $this->password,
            'roles' => $this->roles,
        ));
    }

    public function unserialize(string $data)
    {
        $unserialized = unserialize($data);

        $this->id = $unserialized['id'];
        $this->email = $unserialized['email'];
        $this->password = $unserialized['password'];
        $this->roles = $unserialized['roles'];
    }
}
`

It is good solution? Do you have any other ideas to resolve the problem?

**Comment by priyadi** [comment, marked as answer]
To me, that's still absolutely necessary even without `rekalogika/file`. Without `__serialize` & `__unserialize`, Symfony will serialize the complete `User` entity, along with its entire related entities, to the session, even though the authentication layer only needs the `id`, and it will refresh the entity from the database anyway.

This is not specific to `rekalogika/file`, but applies to all cases where something prevents `User` from being serialized, including other file upload frameworks. But probably a good idea to add to our docs. Thanks.
